lxd/storage/s3/miniod/admin.go

```diff
@@ -0,0 +1,226 @@
+	SecretKey    string    `xml:"SecretAccessKey" json:"secretKey"`
+	SessionToken string    `xml:"SessionToken"    json:"sessionToken"`
+	Expiration   time.Time `xml:"Expiration"      json:"expiration"`
+}
+
+// ServiceAccountArgs is the request options for adding or modifying a service account.
+type ServiceAccountArgs struct {
+	Policy    json.RawMessage `json:"policy"` // Parsed value from iam/policy.Parse()
+	AccessKey string          `json:"accessKey"`
+	SecretKey string          `json:"secretKey"`
+}
+
+// configJSON represents the JSON payload containing the configuration dump.
+type configJSON struct {
+	Status string `json:"status"`
+	Value  []byte `json:"value"`
+}
+
+// NewAdminClient returns a new AdminClient, and assigns the given credentials to an alias, and returns an `mc` client for that alias.
+func NewAdminClient(url string, username string, password string) (*minioAdmin, error) {
+	_, err := shared.RunCommand("mc", "alias", "set", "--insecure", "lxd-minio", api.NewURL().Scheme("http").Host(url).String(), username, password)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to set MinIO client alias: %w", err)
+	}
+
+	return &minioAdmin{alias: "lxd-minio"}, nil
+}
+
+// ServiceStop stops the minio service.
+func (m *minioAdmin) ServiceStop(ctx context.Context) error {
+	_, err := shared.RunCommandContext(ctx, "mc", "admin", "service", "stop", m.alias, "--insecure")
+	if err != nil {
+		return fmt.Errorf("Failed to stop MinIO service: %w", err)
+	}
+
+	return nil
+}
+
+// GetConfig dumps the minio configuration.
+func (m *minioAdmin) GetConfig(ctx context.Context) ([]byte, error) {
+	out, err := shared.RunCommandContext(ctx, "mc", "admin", "config", "export", m.alias, "--json", "--insecure")
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get MinIO config: %w", err)
+	}
+
+	cfg := configJSON{}
+	err = json.Unmarshal([]byte(out), &cfg)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to parse MinIO config: %w", err)
+	}
+
+	return cfg.Value, nil
+}
+
+// ExportIAM exports IAM data and returns a reader to it.
+func (m *minioAdmin) ExportIAM(ctx context.Context) (*zip.Reader, error) {
+	name := "mc"
+	args := []string{"admin", "cluster", "iam", "export", m.alias, "--insecure"}
+	cmd := exec.CommandContext(ctx, name, args...)
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+
+	tmpDir, err := os.MkdirTemp(shared.VarPath("storage-pools"), fmt.Sprintf("%s_iam_export_", m.alias))
+	if err != nil {
+		return nil, err
+	}
+
+	// Once we load the exported zip in memory, we won't need to keep it on disk anymore.
+	defer func() {
+		_ = os.RemoveAll(tmpDir)
+	}()
+
+	// Set the working directory to our tempdir so the zip will be written there.
+	cmd.Dir = tmpDir
+	err = cmd.Run()
+	if err != nil {
+		return nil, shared.NewRunError(name, args, err, &stdout, &stderr)
+	}
+
+	f, err := os.Open(filepath.Join(tmpDir, "iam-info.zip"))
```

Comment:
According to the docs the file it exports to is ALIAS-iam-metadata.zip

https://min.io/docs/minio/linux/reference/minio-mc-admin/mc-admin-cluster-iam-export.html

Reply:
I still need to append the alias, but the actual path is `%s-iam-info.zip`

https://github.com/minio/mc/blob/669cb0a9a475e4f66d837a305800d3eb8ae784a1/cmd/admin-cluster-iam-export.go#L99